File: libs/grpc/clients/websockets.go
package clients

import (
	"log"

	"github.com/twirapp/twir/libs/grpc/constants"
	"github.com/twirapp/twir/libs/grpc/websockets"

	"google.golang.org/grpc"
)

func NewWebsocket(env string) websockets.WebsocketClient {
	serverAddress := createClientAddr(env, "websockets", constants.WEBSOCKET_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		defaultClientsOptions...,
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := websockets.NewWebsocketClient(conn)

	return c
}

File: libs/grpc/clients/websockets.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc'

import { CLIENT_OPTIONS, createClientAddr, waitReady } from './helper'
import { PORTS } from '../constants/constants.js'
import { WebsocketDefinition } from '../websockets/websockets'

import type { WebsocketClient } from '../websockets/websockets'

export async function createWebsocket(env: string): Promise<WebsocketClient> {
	const channel = createChannel(
		createClientAddr(env, 'websockets', PORTS.WEBSOCKET_SERVER_PORT),
		ChannelCredentials.createInsecure(),
		CLIENT_OPTIONS,
	)

	await waitReady(channel)

	const client = createClient(WebsocketDefinition, channel)

	return client as any
}
